### PR TITLE
Added mailchimp example

### PR DESCRIPTION
**Pull request by siowhoi**
Added mailchimp example and unit test
Screenshot 2023-11-29 at 3 27 40 pm

### CODE REVIEW

File: src/examples/mailchimp/.env.example [hunk 1]
```diff
@@ -0,0 +1,4 @@
+MAILCHIMP_API_KEY=
+MAILCHIMP_SERVER_PREFIX=
+MAILCHIMP_LIST_ID=7b86936890
```
Comment (tnodell) [inline src/examples/mailchimp/.env.example:3]: Should these ids be public?

File: src/examples/mailchimp/package.json [hunk 2]
```diff
@@ -0,0 +1,61 @@
+{
+	"name": "@exogee/graphweaver-example-mailchimp",
+	"version": "0.2.6",
```
Comment (tnodell) [inline src/examples/mailchimp/package.json:3]: I believe gw is on 0.2.7 now.

File: src/examples/mailchimp/src/backend/schema/category/data-entity.ts [hunk 3]
```diff
@@ -0,0 +1,14 @@
+import { BaseDataEntity } from '@exogee/graphweaver';
+
+export class CategoryDataEntity implements BaseDataEntity {
```
Comment (tnodell) [inline src/examples/mailchimp/src/backend/schema/category/data-entity.ts:3]: I think of entities in `/schema` as being graphql entities, and these look like base data entities. Is this how AIA does it?

File: src/examples/mailchimp/src/backend/schema/category/entity.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
+import { Field, GraphQLEntity, ID, ObjectType, SummaryField } from '@exogee/graphweaver';
+
+import { CategoryDataEntity } from './data-entity';
+
+@ObjectType()
```
Comment (tnodell) [inline src/examples/mailchimp/src/backend/schema/category/entity.ts:5]: Maybe just split these into 

/entities/category.ts
and
/schema/category.ts

?

This is a stylistic difference so it's up to you. I think we should be consistent with the other examples and keep data entities and schema entities in their own folder.

File: src/examples/mailchimp/src/backend/schema/index.ts [hunk 5]
```diff
@@ -0,0 +1,56 @@
...
+import { InterestResolver, createInterestResolver } from './interest';
+import { MemberResolver, createMemberResolver } from './member';
+
+export type ClientOptions = {
+	apiKey: string;
+	server: string;
+};
+
+export const mailingList: MailingListResolver =
+	process.env.MAILCHIMP_API_KEY &&
```
Comment (tnodell) [inline src/examples/mailchimp/src/backend/schema/index.ts:12]: Would be good to try to catch any of these missing env vars with a specific error.

File: src/examples/mailchimp/src/backend/schema/member/provider.ts [hunk 6]
```diff
@@ -0,0 +1,81 @@
+import Mailchimp from '@mailchimp/mailchimp_marketing/src/ApiClient';
+import {
+	createProvider,
+	ItemWithId,
+	createPaginationOptions,
+	inMemoryFilterFor,
+} from '@exogee/graphweaver-helpers';
+import { ClientOptions } from '../index';
+import { MemberDataEntity, MemberStatus } from './data-entity';
+import md5 from 'md5';
```
Comment (tnodell) [inline src/examples/mailchimp/src/backend/schema/member/provider.ts:10]: I think we use argon2 for encryption. Can we double check that this is this lib to use with @steve-keep ?

File: src/packages/graphweaver-helpers/src/provider/index.ts [hunk 7]
```diff
@@ -82,7 +82,6 @@ export const createProvider = <Entity extends WithId, Context, DataEntity extend
 		}
 
 		_mapDataEntity(dataEntity: DataEntity): DataEntity {
-			console.log('_mapDataEntity ', this?.dataEntity);
```
Comment (tnodell) [inline src/packages/graphweaver-helpers/src/provider/index.ts:85]: ty I think I left this here.

**Review by tnodell — APPROVED** [review]
Couple of small things to check before merging. Otherwise looks good!

File: src/examples/mailchimp/package.json (hunk 2 above)
Comment (siowhoi) [reply src/examples/mailchimp/package.json]: fixed

File: src/packages/graphweaver-helpers/src/provider/index.ts (hunk 7 above)
Comment (siowhoi) [reply src/packages/graphweaver-helpers/src/provider/index.ts:85]: This line has been removed after merging with main.

File: src/examples/mailchimp/src/backend/schema/category/data-entity.ts (hunk 3 above)
Comment (siowhoi) [reply src/examples/mailchimp/src/backend/schema/category/data-entity.ts]: Fixed. They are restructured to entities/category.ts and schema/category

File: src/examples/mailchimp/src/backend/schema/category/entity.ts (hunk 4 above)
Comment (siowhoi) [reply src/examples/mailchimp/src/backend/schema/category/entity.ts:5]: Fixed as suggested.

File: src/examples/mailchimp/src/backend/schema/index.ts (hunk 5 above)
Comment (siowhoi) [reply src/examples/mailchimp/src/backend/schema/index.ts:12]: added check for missing environment variables

File: src/examples/mailchimp/src/backend/schema/member/provider.ts (hunk 6 above)
Comment (siowhoi) [reply src/examples/mailchimp/src/backend/schema/member/provider.ts:10]: Is argon2 the same as md5? Mailchimp require md5 hashing on the email field.

**Pull request merged**